StRoot/StFstHitMaker/StFstHitMaker.cxx

```diff
@@ -0,0 +1,214 @@
+				newHit->setId(key);
+				newHit->setIdTruth(idTruth);
+
+                                int moduleIdx;
+				if(disk == 1) // Disk 1
+					moduleIdx = wedge;
+				if(disk == 2)// Disk 2
+					moduleIdx = wedge-12;
+				if(disk == 3)// Disk 3
+					moduleIdx = wedge-24;
+				if(disk == 1 || disk == 3)
+				{// Disk 1 & 3
+					phiInner = kFstphiStart[moduleIdx-1]*TMath::Pi()/6.0 + 0.5*kFstzDirct[moduleIdx-1]*kFstStripPitchPhi;
+					phiOuter = kFstphiStop[moduleIdx-1]*TMath::Pi()/6.0  - 0.5*kFstzDirct[moduleIdx-1]*kFstStripPitchPhi;
+				}
+				if(disk == 2)
+				{ // Disk 2
+					phiInner = kFstphiStop[moduleIdx-1]*TMath::Pi()/6.0  - 0.5*kFstzDirct[moduleIdx-1]*kFstStripPitchPhi;
+					phiOuter = kFstphiStart[moduleIdx-1]*TMath::Pi()/6.0 + 0.5*kFstzDirct[moduleIdx-1]*kFstStripPitchPhi;
+				}
+				double local[3];
+				if(meanRStrip < kFstNumRStripsPerWedge/2)
+				{ // inner
+					local[0] = kFstrStart[meanRStrip] + 0.5*kFstStripPitchR;
+					local[1] = phiInner + kFstzFilp[disk-1]*kFstzDirct[moduleIdx-1]*meanPhiStrip*kFstStripPitchPhi;
+				}
+				else
+				{// outer
+					if(sensor == 1){
+						local[0] = kFstrStart[meanRStrip] + 0.5*kFstStripPitchR;
+						local[1] = phiOuter - kFstzFilp[disk-1]*kFstzDirct[moduleIdx-1]*meanPhiStrip*kFstStripPitchPhi - kFstzFilp[disk-1]*kFstzDirct[moduleIdx-1]*0.5*kFstStripGapPhi;
+					}
+					if(sensor == 2){
+						local[0] = kFstrStart[meanRStrip] + 0.5*kFstStripPitchR;
+						local[1] = phiOuter - kFstzFilp[disk-1]*kFstzDirct[moduleIdx-1]*meanPhiStrip*kFstStripPitchPhi + kFstzFilp[disk-1]*kFstzDirct[moduleIdx-1]*0.5*kFstStripGapPhi;
+					}
+				}
+
+				if(disk == 1) local[2] = 146.6012; //unit: cm
+				if(disk == 2) local[2] = 160.1724; //unit: cm
+				if(disk == 3) local[2] = 173.7436; //unit: cm
+				newHit->setLocalPosition(local[0], local[1], local[2]); //set local position on sensor
+
+				fstHitCollection->addHit(newHit);
+			} //cluster loop over
+		}//end clusterCollectionPtr
+
+		//set global position
+		StFstWedgeHitCollection *wedgeHitCollection = fstHitCollection->wedge(wedgeIdx);
+		for (int sensorIdx = 0; sensorIdx < kFstNumSensorsPerWedge; sensorIdx++) {
+			StFstSensorHitCollection *sensorHitCollection = wedgeHitCollection->sensor(sensorIdx);
+
+			for (int idx = 0; idx < (int) sensorHitCollection->hits().size(); idx++ ) {
+				StFstHit *newHit = sensorHitCollection->hits()[idx];
+				double local[3];
+				double global[3];
+				local[0] = newHit->localPosition(0);
+				local[1] = newHit->localPosition(1);
+				local[2] = newHit->localPosition(2);
+
+                                //Ye: simple transformation. Need to revisit
+				int sensorId = 1000 + ((int)newHit->getWedge() - 1) * kFstNumSensorsPerWedge + (int)newHit->getSensor();
+				TGeoHMatrix *geoMSensorOnGlobal = (TGeoHMatrix *) mSensorTransforms->FindObject(Form("R%04i", sensorId));
+				geoMSensorOnGlobal->LocalToMaster(local, global);
+
```

Comment:
seems like testing shortcut w/o any table entries.

Reply:
The geometry table for FST is not ready to be uploaded to the database. For now, we would prefer to use hard-coded number for the geometry.
We will update the code to read in the geometry tables later.

Reply:
as we discussed. Please add comments in the cod where you want to update with actual tables.
The state Ye... is note sufficient. I am ok with such an addition to make clear where propoer changes has to be done.
By the way the position for disk in z are not the one deduced from the survey as documented

Reply:
added comment that the geometry table will be used later